src/plugins/intel_cpu/src/transformations/cpu_opset/x64/pass/convert_precision_i64_i32.cpp

```diff
@@ -0,0 +1,105 @@
+        } else {
+            dstData[i] = static_cast<int32_t>(srcData[i]);
+        }
+    }
+    return newConstant;
+}
+
+bool ov::intel_cpu::ConvertPrecisionI64ToI32::run_on_model(const std::shared_ptr<ov::Model> &model) {
+    const auto orderedOps = model->get_ordered_ops();
+    for (const auto& op : orderedOps) {
+        if (isNativelySupported(op->get_type_info()) || TypeFromName(op->get_type_name()) == Type::Unknown) {
+            continue;
+        }
+
+        bool convertForOutputsRequired = false;
+        for (const auto& input : op->inputs()) {
+            if (input.get_element_type() == ov::element::i64) {
+                auto parentOutput = input.get_source_output();
+                auto parentNode = parentOutput.get_node_shared_ptr();
+                if (ov::is_type<ov::opset12::Convert>(parentNode) &&
+                        parentNode->get_rt_info().find("convert_i32_i64") != parentNode->get_rt_info().end()) {
+                    input.replace_source_output(parentNode->input_value(0));
+                } else if (auto constOp = ov::as_type_ptr<ov::op::v0::Constant>(parentNode)) {
+                    auto newConst = changeConstantPrecision(constOp);
+                    input.replace_source_output(newConst);
+                    newConst->set_friendly_name(constOp->get_friendly_name());
+                } else {
+                    auto convert = std::make_shared<ov::opset12::Convert>(input.get_source_output(), ov::element::i32);
+                    convert->output(0).add_names(parentOutput.get_names());
+                    input.replace_source_output(convert);
+                }
+                convertForOutputsRequired = true;
+            }
+        }
+
+        if (convertForOutputsRequired) {
+            // Propagate i32 precision into outputs.
+            op->validate_and_infer_types();
+            for (auto& output : op->outputs()) {
+                if (output.get_element_type() == ov::element::i32) {
+                    auto targetInputs = output.get_target_inputs();
+                    auto convert = std::make_shared<ov::opset12::Convert>(output, ov::element::i64);
+
+                    auto& rt_info = convert->get_rt_info();
+                    rt_info["convert_i32_i64"] = "";
+                    for (const auto& targetInput : targetInputs) {
+                        targetInput.replace_source_output(convert);
+                    }
+
+                    auto& convertTensor = convert->output(0).get_tensor();
+                    const std::string newName = ov::op::util::get_ie_output_name(output);
+                    if (ov::descriptor::get_ov_tensor_legacy_name(convertTensor).empty()) {
+                        ov::descriptor::set_ov_tensor_legacy_name(convertTensor, newName);
+                    }
+                    if (!output.get_names().empty()) {
+                        convertTensor.set_names(output.get_names());
+                    }
```

Comment:
I suppose we can use `ov::replace_output_update_name` instead of this logic

Reply:
Done